rpc/client.go

```diff
@@ -265,18 +265,19 @@ func (c *Client) heartbeat() error {
 				remoteTimeNow := response.ServerTime + (receiveTime-sendTime)/2
 				c.offset.Offset = remoteTimeNow - receiveTime
 			}
+			offset := c.offset
```

Comment:
these changes are unrelated to the description in the commit message - could they be in a separate commit?